ts/deck-options/DisplayOrder.svelte

```diff
@@ -43,6 +43,32 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
         tr.deckConfigSortOrderAscendingEase(),
         tr.deckConfigSortOrderDescendingEase(),
     ];
+
+    let disabledNewSortOrders: number[] = [];
+    $: {
+        switch ($config.newCardGatherPriority) {
+            case 3:
+                // random notes
+                // - 'template, gather' is the same as 'template, random' (2)
+                // - 'gather' is the same as 'random note, template' (3)
+                disabledNewSortOrders = [2, 3];
+                break;
+            case 4:
+                // random cards
+                // - 'template, gather' is the same as 'template, random' (2)
+                // - 'random note, template' is not useful if siblings are not
+                //    gathered together (3)
+                // - 'gather' is the same as 'random' (4)
+                disabledNewSortOrders = [2, 3, 4];
+                break;
+            default:
+                disabledNewSortOrders = [];
+                break;
+        }
+        if (disabledNewSortOrders.includes($config.newCardSortOrder)) {
+            $config.newCardSortOrder = 0;
+        }
+    }
```

Comment:
Sorry, I should have been clearer. What gives me trouble are the choices arrays. I'd like to associate every variant with exactly one string, and get a type error if one variant is missing (or a non-exisiting one referenced of course). I did a bit of research and there seems to be the [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) type, but it doesn't seem to work with enums in our TS version.

Makes sense to still use enums directly where possible, of course, and I'll commit your diff shortly.

Reply:
Gotcha. I imagine it might be possible if we switched the choices array into a record, and used something like keyof. @hgiesel might know more. No need to hold up this PR over it though. Run out of time tonight, but will give this a final review tomorrow morning. :-)